src/lib/add-on/clients/add-on_auto.rb

```diff
@@ -1,328 +1,319 @@
-      # return boolean
-      #
-      #
-      # **Structure:**
-      #
-      #
-      #      <add-on>
-      #     	<add_on_products config:type="list">
-      #     		<listentry>
-      #     			<media_url>http://software.opensuse.org/download/server:/dns/SLE_10/</media_url>
-      #     			<product>buildservice</product>
-      #     			<product_dir>/</product_dir>
-      #     			<!-- (optional) -->
-      #     			<name>User-Defined Product Name</name>
-      #     			<signature-handling>
-      #     				<accept_unsigned_file config:type="boolean">true</accept_unsigned_file>
-      #     				<accept_file_without_checksum config:type="boolean">true</accept_file_without_checksum>
-      #     				<accept_verification_failed config:type="boolean">true</accept_verification_failed>
-      #     				<accept_unknown_gpg_key>
-      #     					<all config:type="boolean">true</all>
-      #     					<keys config:type="list">
-      #     						<keyid>...</keyid>
-      #     						<keyid>3B3011B76B9D6523</keyid>
-      #     					</keys>
-      #     				</accept_unknown_gpg_key>
-      #     				<accept_non_trusted_gpg_key>
-      #     				<all config:type="boolean">true</all>
-      #     					<keys config:type="list">
-      #     						<keyid>...</keyid>
-      #     					</keys>
-      #     				</accept_non_trusted_gpg_key>
-      #     				<import_gpg_key>
-      #     					<all config:type="boolean">true</all>
-      #     					<keys config:type="list">
-      #     						<keyid>...</keyid>
-      #     					</keys>
-      #     				</import_gpg_key>
-      #     			</signature-handling>
-      #     		</listentry>
-      #     	</add_on_products>
-      #      </add-on>
-      #
-      elsif @func == "Write"
-        @sources = {}
-
-        AddOnProduct.add_on_products.each do |prod|
-          media = Ops.get_string(prod, "media_url", "")
-          pth = Ops.get_string(prod, "product_dir", "/")
-          if String.StartsWith(media, "relurl://")
-            base = AddOnProduct.GetBaseProductURL
-            media = AddOnProduct.GetAbsoluteURL(base, media)
-            Builtins.y2milestone("relurl changed to %1", media)
-          end
-          Ops.set(@sources, media, Ops.get(@sources, media, {}))
-          # set addon specific sig-handling
-          AddOnProduct.SetSignatureCallbacks(
-            Ops.get_string(prod, "product", "")
-          )
-          srcid = -1
-          begin
-            url = AddOnProduct.SetRepoUrlAlias(
-              # Expanding URL in order to "translate" tags like $releasever
-              Pkg.ExpandedUrl(media),
-              Ops.get_string(prod, "alias", ""),
-              Ops.get_string(prod, "name", "")
-            )
-
-            srcid = Pkg.SourceCreate(url, pth)
-
-            if (srcid == -1 || srcid == nil)
-              # revert back to the unexpanded URL to have the original URL
-              # in the saved /etc/zypp/repos.d file
-              Pkg.SourceChangeUrl(srcid, media)
```

Comment:
Reading the source code of [`Pkg.SourceChangeUrl`](https://github.com/yast/yast-pkg-bindings/blob/af628652aa989802c6acd60ca4714b8928ab15f2/src/Source_Set.cc#L269-L271), I think that this line was doing nothing. Am I right?